### PR TITLE
[docathon] Add examples for isdiff/isdirty

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
Nice examples, I like it 👍

**Comment by kshyatt** [comment]
These examples are "ripped from the headlines" of `test/libgit2.jl` if anyone would like to do something similar.

File: base/libgit2/libgit2.jl [hunk 1]
```diff
@@ -95,6 +95,17 @@ Checks if there have been any changes to tracked files in the working tree (if
 `cached=false`) or the index (if `cached=true`).
 `pathspecs` are the specifications for options for the diff.
 
+# Example
+```julia
+repo = LibGit2.GitRepo(repo_path)
+LibGit2.isdirty(repo) #should be false
```
Comment (tkelman) [inline base/libgit2/libgit2.jl]: (I'd just edit, but you might disagree)
I think comments are a little more readable with a space after the `#`

Comment (kshyatt) [inline base/libgit2/libgit2.jl]: I really don't have an opinion so if you feel strongly go ahead and edit

Comment (kshyatt) [inline base/libgit2/libgit2.jl]: @tkelman you want to edit or should we just merge?

**Pull request merged**